lago/subnet_lease.py

```diff
@@ -17,222 +17,505 @@
+    the lease files in the store (each file will be represented with a Lease
+    object).
+
+    Cleanup of stale leases is done in a lazy manner during a request for a
+    lease. The store will remove at most 1 stale lease in each request (see
+    SubnetStore._lease_valid for more info).
+
+    Attributes:
+        _path (str): Path to the store, if not specified defaults to the value
+            of 'lease_dir' in the config
+        _cidr (int): Number of bits dedicated for the network address.
+            Has a fixed value of 24.
+        _subnet_template (str): A template for creating ip address.
+            Has a fixed value of '192.168.{}.0'
+        _min_third_octet (int): The minimum value of the subnets' last octet.
+        _max_third_octet (int): The maximum value of the subnets' last octet.
+        _min_subnet (netaddr.IPNetwork): The lowest subnet in the range of
+            the store.
+        _max_subnet (netaddr.IPNetwork): The highest subnet in the range of
+            the store.
     """
-    Checs if the given lease still has a prefix that owns it
-
-    Args:
-        path (str): Path to the lease
-
-    Returns:
-        bool: ``True`` if the uuid path in the lease still exists and is the
-            same as the one in the lease
-    """
-    with open(path) as f:
-        uuid_path, uuid = json.load(f)
-
-    if not os.path.isfile(uuid_path):
-        return False
-
-    with open(uuid_path) as f:
-        return f.read() == uuid
 
-
-@_validate_lease_dir_present
-@_locked
-def _acquire(uuid_path):
-    """
-    Lease a free network for the given uuid path
-
-    Args:
-        uuid_path (str): Path to the uuid file of a :class:`lago.Prefix`
-
-    Returns:
-        int or None: the third element of the dotted ip of the leased network
-            or ``None`` if no lease was available
-
-    .. todo::
-        Raise exception or something instead of returning None so the
-        caller can handle the failure case
-    """
-
-    lease_dir = config['lease_dir']
-    for index in range(MIN_SUBNET, MAX_SUBNET + 1):
-        lease_file = os.path.join(lease_dir, '%d.lease' % index)
-        if os.path.exists(lease_file):
-            if _lease_valid(lease_file):
+    def __init__(
+        self,
+        path=None,
+        min_third_octet=200,
+        max_third_octet=255,
+    ):
+
+        self._path = path or config['lease_dir']
+        self._cidr = 24
+        self._subnet_template = '{}/{}'.format('192.168.{}.0', self._cidr)
+        self._min_third_octet = min_third_octet
+        self._max_third_octet = max_third_octet
+        self._min_subnet = IPNetwork(
+            self._subnet_template.format(min_third_octet)
+        )
+        self._max_subnet = IPNetwork(
+            self._subnet_template.format(max_third_octet)
+        )
+        self._validate_lease_dir()
+
+    def _create_lock(self):
+        return utils.LockFile(
+            path=os.path.join(self.path, LOCK_NAME), timeout=5
```

Comment:
LockFile will take care of retries in intervals of 0.1 second (this value is hardcoded).